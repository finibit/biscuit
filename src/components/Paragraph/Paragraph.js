import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import styles from '../../themes'

const ParagraphStyled = styled.p`
	${styles.fontFamily}
	${styles.fontSize}
	${styles.textAlign}
	${styles.spacing}
	${styles.color}

	font-weight: normal;

	${(props) => props.$css}
`

/** A paragraph. */
const Paragraph = (props) => {
	const {
		color,
		align,
		size,
		margin,
		padding,
		themeElement,
		css,
		...rest
	} = props

	return (
		<ParagraphStyled
			$element={themeElement}
			$color={color}
			$align={align}
			$size={size}
			$margin={margin}
			$padding={padding}
			$css={css}
			{...rest}
		/>
	)
}

Paragraph.propTypes = {
	/** Any number of renderable nodes. */
	children: PropTypes.node,

	/** Text color. */
	color: PropTypes.oneOfType([
		PropTypes.number,
		PropTypes.string,
		PropTypes.shape({
			value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
			shade: PropTypes.number,
			tint: PropTypes.number,
		}),
	]),

	/** Text alignment. */
	align: PropTypes.oneOf(['left', 'center', 'right']),

	/** Size of the paragraph. */
	size: PropTypes.oneOfType([
		PropTypes.number,
		PropTypes.string,
		PropTypes.object,
		PropTypes.array,
	]),

	/** The amount of margin around the paragraph. */
	margin: PropTypes.oneOfType([
		PropTypes.number,
		PropTypes.string,
		PropTypes.object,
		PropTypes.array,
	]),

	/** The amount of padding around the paragraph. */
	padding: PropTypes.oneOfType([
		PropTypes.number,
		PropTypes.string,
		PropTypes.object,
		PropTypes.array,
	]),

	/** Theme element. */
	themeElement: PropTypes.string,

	/** Custom styles passed to styled-components. */
	css: PropTypes.oneOfType([PropTypes.string, PropTypes.object, PropTypes.array]),
}

Paragraph.defaultProps = {
	children: null,
	color: 0,
	align: 'left',
	size: 2,
	margin: { bottom: 0 },
	padding: 'none',
	themeElement: 'Paragraph',
	css: null,
}

export default Paragraph
