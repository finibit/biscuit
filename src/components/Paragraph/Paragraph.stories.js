import React from 'react'
import { storiesOf } from '@storybook/react'

import { Theme, Paragraph } from '../../../src'

storiesOf('Paragraph', module)
	.add('default', () => (
		<Theme>
			<Paragraph>Finite but unbounded vastness is bearable only through love star stuff harvesting star light rogue laws of physics a billion trillion? Network of wormholes at the edge of forever the only home we've ever known billions upon billions white dwarf the sky calls to us. Permanence of the stars bits of moving fluff courage of our questions hearts of the stars citizens of distant epochs Cambrian explosion. Emerged into consciousness invent the universe inconspicuous motes of rock and gas globular star cluster descended from astronomers concept of the number one and billions upon billions upon billions upon billions upon billions upon billions upon billions.</Paragraph>
		</Theme>
	))
	.add('color', () => (
		<Theme>
			<Paragraph color="light-4">Ship of the imagination from which we spring courage of our questions as a patch of light vanquish the impossible concept of the number one. Network of wormholes not a sunrise but a galaxyrise laws of physics how far away invent the universe the only home we've ever known. Network of wormholes made in the interiors of collapsing stars emerged into consciousness encyclopaedia galactica the sky calls to us trillion.</Paragraph>
			<Paragraph color="#caaffe">Invent the universe with pretty stories for which there's little good evidence hearts of the stars birth quasar gathered by gravity. Rich in heavy atoms as a patch of light concept of the number one rich in heavy atoms another world not a sunrise but a galaxyrise. Are creatures of the cosmos dispassionate extraterrestrial observer a still more glorious dawn awaits two ghostly white figures in coveralls and helmets are soflty dancing concept of the number one made in the interiors of collapsing stars and billions upon billions upon billions upon billions upon billions upon billions upon billions.</Paragraph>
		</Theme>
	))
	.add('align', () => (
		<Theme>
			<Paragraph align="left">Ship of the imagination from which we spring courage of our questions as a patch of light vanquish the impossible concept of the number one. Network of wormholes not a sunrise but a galaxyrise laws of physics how far away invent the universe the only home we've ever known. Network of wormholes made in the interiors of collapsing stars emerged into consciousness encyclopaedia galactica the sky calls to us trillion.</Paragraph>
			<Paragraph align="right">Invent the universe with pretty stories for which there's little good evidence hearts of the stars birth quasar gathered by gravity. Rich in heavy atoms as a patch of light concept of the number one rich in heavy atoms another world not a sunrise but a galaxyrise. Are creatures of the cosmos dispassionate extraterrestrial observer a still more glorious dawn awaits two ghostly white figures in coveralls and helmets are soflty dancing concept of the number one made in the interiors of collapsing stars and billions upon billions upon billions upon billions upon billions upon billions upon billions.</Paragraph>
			<Paragraph align="center">Ship of the imagination from which we spring courage of our questions as a patch of light vanquish the impossible concept of the number one. Network of wormholes not a sunrise but a galaxyrise laws of physics how far away invent the universe the only home we've ever known. Network of wormholes made in the interiors of collapsing stars emerged into consciousness encyclopaedia galactica the sky calls to us trillion.</Paragraph>
		</Theme>
	))
	.add('size', () => (
		<Theme>
			<Paragraph size="xs">Brain is the seed of intelligence paroxysm of global death two ghostly white figures in coveralls and helmets are soflty dancing are creatures of the cosmos the carbon in our apple pies courage of our questions. The sky calls to us as a patch of light ship of the imagination vanquish the impossible a very small stage in a vast cosmic arena gathered by gravity.</Paragraph>
			<Paragraph size="sm">Brain is the seed of intelligence paroxysm of global death two ghostly white figures in coveralls and helmets are soflty dancing are creatures of the cosmos the carbon in our apple pies courage of our questions. The sky calls to us as a patch of light ship of the imagination vanquish the impossible a very small stage in a vast cosmic arena gathered by gravity.</Paragraph>
			<Paragraph size="md">Brain is the seed of intelligence paroxysm of global death two ghostly white figures in coveralls and helmets are soflty dancing are creatures of the cosmos the carbon in our apple pies courage of our questions. The sky calls to us as a patch of light ship of the imagination vanquish the impossible a very small stage in a vast cosmic arena gathered by gravity.</Paragraph>
			<Paragraph size="lg">Brain is the seed of intelligence paroxysm of global death two ghostly white figures in coveralls and helmets are soflty dancing are creatures of the cosmos the carbon in our apple pies courage of our questions. The sky calls to us as a patch of light ship of the imagination vanquish the impossible a very small stage in a vast cosmic arena gathered by gravity.</Paragraph>
			<Paragraph size="xl">Brain is the seed of intelligence paroxysm of global death two ghostly white figures in coveralls and helmets are soflty dancing are creatures of the cosmos the carbon in our apple pies courage of our questions. The sky calls to us as a patch of light ship of the imagination vanquish the impossible a very small stage in a vast cosmic arena gathered by gravity.</Paragraph>
		</Theme>
	))
	.add('size responsive', () => (
		<Theme>
			<Paragraph size={['md', 'md', 'lg', 'xl']}>Brain is the seed of intelligence paroxysm of global death two ghostly white figures in coveralls and helmets are soflty dancing are creatures of the cosmos the carbon in our apple pies courage of our questions. The sky calls to us as a patch of light ship of the imagination vanquish the impossible a very small stage in a vast cosmic arena gathered by gravity.</Paragraph>
		</Theme>
	))
	.add('padding', () => (
		<Theme>
			<Paragraph padding="md" css="background-color: violet;">Radio telescope dispassionate extraterrestrial observer Cambrian explosion hearts of the stars paroxysm of global death vastness is bearable only through love? White dwarf circumnavigated not a sunrise but a galaxyrise encyclopaedia galactica are creatures of the cosmos finite but unbounded.</Paragraph>
		</Theme>
	))
	.add('padding responsive', () => (
		<Theme>
			<Paragraph padding={['none', 'xs', 'sm', 'md']} css="background-color: violet;">Radio telescope dispassionate extraterrestrial observer Cambrian explosion hearts of the stars paroxysm of global death vastness is bearable only through love? White dwarf circumnavigated not a sunrise but a galaxyrise encyclopaedia galactica are creatures of the cosmos finite but unbounded.</Paragraph>
		</Theme>
	))
	.add('margin', () => (
		<Theme>
			<Paragraph margin="md" css="background-color: violet;">Radio telescope dispassionate extraterrestrial observer Cambrian explosion hearts of the stars paroxysm of global death vastness is bearable only through love? White dwarf circumnavigated not a sunrise but a galaxyrise encyclopaedia galactica are creatures of the cosmos finite but unbounded.</Paragraph>
			<Paragraph margin="md" css="background-color: violet;">Dream of the mind's eye kindling the energy hidden in matter brain is the seed of intelligence two ghostly white figures in coveralls and helmets are soflty dancing from which we spring with pretty stories for which there's little good evidence and billions upon billions upon billions upon billions upon billions upon billions upon billions.</Paragraph>
		</Theme>
	))
	.add('margin responsive', () => (
		<Theme>
			<Paragraph margin={['none', 'xs', 'sm', 'md']} css="background-color: violet;">Radio telescope dispassionate extraterrestrial observer Cambrian explosion hearts of the stars paroxysm of global death vastness is bearable only through love? White dwarf circumnavigated not a sunrise but a galaxyrise encyclopaedia galactica are creatures of the cosmos finite but unbounded.</Paragraph>
			<Paragraph margin={['none', 'xs', 'sm', 'md']} css="background-color: violet;">Dream of the mind's eye kindling the energy hidden in matter brain is the seed of intelligence two ghostly white figures in coveralls and helmets are soflty dancing from which we spring with pretty stories for which there's little good evidence and billions upon billions upon billions upon billions upon billions upon billions upon billions.</Paragraph>
		</Theme>
	))
