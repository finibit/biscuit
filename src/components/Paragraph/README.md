### Usage

#### Color

```jsx
<Paragraph color="info">The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Paragraph>
```
```jsx
<Paragraph color="#009900">The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Paragraph>
```
```jsx
<Paragraph color="violet">The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Paragraph>
```

#### Align

```jsx
<Paragraph as="div" align="left">The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Paragraph>
```
```jsx
<Paragraph as="div" align="right">The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Paragraph>
```
```jsx
<Paragraph as="div" align="center">The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Paragraph>
```

#### Size

```jsx
<Paragraph size={4}>The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Paragraph>
```
```jsx
<Paragraph size="2rem">The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Paragraph>
```
```jsx
<Paragraph size={[0, 1, 2, 3]}>The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Paragraph>
```

#### Space

```jsx
<Paragraph as="div" margin="10px" padding="10px">The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Paragraph>
```

```jsx
<Paragraph as="div" margin={2} padding={2}>The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Paragraph>
```

```jsx
<Paragraph as="div" margin={[0, 1, 2, 3]} padding={[0, 1, 2, 3]}>The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Paragraph>
```

#### Css

```jsx
<Paragraph css="border 1px solid #000;">The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Paragraph>
```

```jsx
<Paragraph css={{ border: "1px solid #000;" }}>The sky calls to us Tunguska event cosmic fugue hydrogen atoms prime number take root and flourish. Made in the interiors of collapsing stars hearts of the stars across the centuries concept of the number one realm of the galaxies the only home we've ever known.</Paragraph>
```

```jsx
import { css } from 'styled-components';

const style = css`
  color: ${(props) => props.theme.global.colors.info};
  border: 1px solid ${(props) => props.theme.global.colors.info};
`;

<Paragraph css={style}>Paragraph</Paragraph>
```
