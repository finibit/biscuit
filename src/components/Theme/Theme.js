import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider, createGlobalStyle } from 'styled-components'
import { mergeTheme, baseTheme, themeGet } from '../../themes'
import types from '../../types'

const GlobalStyle = createGlobalStyle`
	html, body {
		width: 100%;
		height: 100%;
		margin: 0;
		padding: 0;
	}

	body {
		color: ${themeGet.color('black', 'inherit')};
		font-family: ${themeGet.fontFamily('secondary')};
		-webkit-font-smoothing: antialiased;
	}
`

const Theme = (props) => {
	const {
		theme,
		global,
	} = props

	return (
		<ThemeProvider theme={mergeTheme(baseTheme, theme)}>
			<div>
				{ global && <GlobalStyle /> }
				{props.children}
			</div>
		</ThemeProvider>
	)
}

Theme.propTypes = {
	global: PropTypes.bool,
	...types.theme,
}

Theme.defaultProps = {
	global: true,
	theme: baseTheme,
}

export default Theme
