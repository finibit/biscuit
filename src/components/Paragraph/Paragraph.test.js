import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Theme from '../Theme'
import Paragraph from '../Paragraph'

describe('Paragraph', () => {
  test('default', () => {
    const component = renderer.create((
      <Theme>
        <Paragraph />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('color', () => {
    const component = renderer.create((
      <Theme>
        <Paragraph color='light-0' />
        <Paragraph color='#aabbcc' />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('align', () => {
    const component = renderer.create((
      <Theme>
        <Paragraph align='left' />
        <Paragraph align='right' />
        <Paragraph align='center' />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('size', () => {
    const component = renderer.create((
      <Theme>
        <Paragraph size='xs' />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('size responsive', () => {
    const component = renderer.create((
      <Theme>
        <Paragraph size={['xs', 'sm', 'md', 'lg']} />
        <Paragraph size={{ xs: 'xs', sm: 'sm', md: 'md', lg: 'lg' }} />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('padding', () => {
    const component = renderer.create((
      <Theme>
        <Paragraph padding='xs' />
        <Paragraph padding={{}} />
        <Paragraph padding={{ top: 'xs' }} />
        <Paragraph padding={{ right: 'xs' }} />
        <Paragraph padding={{ bottom: 'xs' }} />
        <Paragraph padding={{ left: 'xs' }} />
        <Paragraph padding={{ horizontal: 'xs' }} />
        <Paragraph padding={{ vertical: 'xs' }} />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('padding responsive', () => {
    const component = renderer.create((
      <Theme>
        <Paragraph padding={['xs', 'sm', 'md', 'lg']} />
        <Paragraph padding={{ xs: 'xs', sm: 'sm', md: 'md', lg: 'lg' }} />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('margin', () => {
    const component = renderer.create((
      <Theme>
        <Paragraph margin='xs' />
        <Paragraph margin={{}} />
        <Paragraph margin={{ top: 'xs' }} />
        <Paragraph margin={{ right: 'xs' }} />
        <Paragraph margin={{ bottom: 'xs' }} />
        <Paragraph margin={{ left: 'xs' }} />
        <Paragraph margin={{ horizontal: 'xs' }} />
        <Paragraph margin={{ vertical: 'xs' }} />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('margin responsive', () => {
    const component = renderer.create((
      <Theme>
        <Paragraph margin={['xs', 'sm', 'md', 'lg']} />
        <Paragraph margin={{ xs: 'xs', sm: 'sm', md: 'md', lg: 'lg' }} />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })

  test('css', () => {
    const component = renderer.create((
      <Theme>
        <Paragraph css='background-color: #000;' />
        <Paragraph css={{ backgroundColor: '#000' }} />
      </Theme>
    ))
    expect(component.toJSON()).toMatchSnapshot()
  })
})
