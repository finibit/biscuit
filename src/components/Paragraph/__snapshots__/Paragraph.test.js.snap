// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph align 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: right;
  margin: 0 0 0.402rem 0;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

.c2 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: center;
  margin: 0 0 0.402rem 0;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

<div>
  <p
    className="c0"
  />
  <p
    className="c1"
  />
  <p
    className="c2"
  />
</div>
`;

exports[`Paragraph color 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: 0;
  color: light-0;
  font-weight: normal;
}

.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: 0;
  color: #aabbcc;
  font-weight: normal;
}

<div>
  <p
    className="c0"
  />
  <p
    className="c1"
  />
</div>
`;

exports[`Paragraph css 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: 0;
  color: #000000;
  font-weight: normal;
  background-color: #000;
}

<div>
  <p
    className="c0"
  />
  <p
    className="c0"
  />
</div>
`;

exports[`Paragraph default 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

<div>
  <p
    className="c0"
  />
</div>
`;

exports[`Paragraph margin 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: xs;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: initial;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

.c2 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: xsrem 0 0 0;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

.c3 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 xsrem 0 0;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

.c4 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 xsrem 0;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

.c5 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0 xsrem;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

.c6 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 xsrem 0 xsrem;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

.c7 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: xsrem 0 xsrem 0;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

<div>
  <p
    className="c0"
  />
  <p
    className="c1"
  />
  <p
    className="c2"
  />
  <p
    className="c3"
  />
  <p
    className="c4"
  />
  <p
    className="c5"
  />
  <p
    className="c6"
  />
  <p
    className="c7"
  />
</div>
`;

exports[`Paragraph margin responsive 1`] = `
.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: initial;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: xs;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

@media only screen and (min-width:48rem) {
  .c0 {
    margin: sm;
  }
}

@media only screen and (min-width:64rem) {
  .c0 {
    margin: md;
  }
}

@media only screen and (min-width:75rem) {
  .c0 {
    margin: lg;
  }
}

<div>
  <p
    className="c0"
  />
  <p
    className="c1"
  />
</div>
`;

exports[`Paragraph padding 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: xs;
  color: #000000;
  font-weight: normal;
}

.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: initial;
  color: #000000;
  font-weight: normal;
}

.c2 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: xsrem 0 0 0;
  color: #000000;
  font-weight: normal;
}

.c3 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: 0 xsrem 0 0;
  color: #000000;
  font-weight: normal;
}

.c4 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: 0 0 xsrem 0;
  color: #000000;
  font-weight: normal;
}

.c5 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: 0 0 0 xsrem;
  color: #000000;
  font-weight: normal;
}

.c6 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: 0 xsrem 0 xsrem;
  color: #000000;
  font-weight: normal;
}

.c7 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: xsrem 0 xsrem 0;
  color: #000000;
  font-weight: normal;
}

<div>
  <p
    className="c0"
  />
  <p
    className="c1"
  />
  <p
    className="c2"
  />
  <p
    className="c3"
  />
  <p
    className="c4"
  />
  <p
    className="c5"
  />
  <p
    className="c6"
  />
  <p
    className="c7"
  />
</div>
`;

exports[`Paragraph padding responsive 1`] = `
.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: initial;
  color: #000000;
  font-weight: normal;
}

.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: 1rem;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: xs;
  color: #000000;
  font-weight: normal;
}

@media only screen and (min-width:48rem) {
  .c0 {
    padding: sm;
  }
}

@media only screen and (min-width:64rem) {
  .c0 {
    padding: md;
  }
}

@media only screen and (min-width:75rem) {
  .c0 {
    padding: lg;
  }
}

<div>
  <p
    className="c0"
  />
  <p
    className="c1"
  />
</div>
`;

exports[`Paragraph size 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: xs;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

<div>
  <p
    className="c0"
  />
</div>
`;

exports[`Paragraph size responsive 1`] = `
.c0 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: xs;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

.c1 {
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
  font-size: initial;
  text-align: left;
  margin: 0 0 0.402rem 0;
  padding: 0;
  color: #000000;
  font-weight: normal;
}

@media only screen and (min-width:48rem) {
  .c0 {
    font-size: sm;
  }
}

@media only screen and (min-width:64rem) {
  .c0 {
    font-size: md;
  }
}

@media only screen and (min-width:75rem) {
  .c0 {
    font-size: lg;
  }
}

<div>
  <p
    className="c0"
  />
  <p
    className="c1"
  />
</div>
`;
